outlines/generate/cfg.py

```diff
@@ -31,6 +33,24 @@ def cfg(model, cfg_str: str, sampler: Sampler = multinomial()) -> SequenceGenera
     return generator
 
 
+@cfg.register(LlamaCpp)
+def cfg_llamacpp(
+    model: LlamaCpp,
+    cfg_str: str,
+    sampler: Sampler = multinomial(),
+):
+    if not isinstance(sampler, multinomial):
+        raise NotImplementedError(
+            r"The llama.cpp integration does not currently support any other sampling algorithm "
+            + "that the multinomial sampler."
```

Comment:
nit: "than"

Reply:
We should probably remove this alltogether because llama.cpp does not use the provided sampler at all.

Reply:
We can pass kwargs related to sampling to `LlamaCpp` based on the sampler passed by the user.  We’re trying to build a unified interface where models are interchangeable 🙂

Reply:
Do you want this to be handled in this PR? The conversion logic does not seem to be a small feat IMHO.